File: C++/Medium/RuntimeError/exclusive-time-of-functions_5.cpp
class Solution {
public:
    vector<int> exclusiveTime(int n, vector<string>& logs) {
        
        vector<int> result(n, 0);
        stack<int> s;
        int prevTime = 0;
        for (auto &log: logs) {
            int pos = log.find(':');
            int id = stoi(log.substr(0, pos));
            int time = stoi(log.substr(pos+2));
            if (log[pos+1] == 's') {
                if (!s.empty()) {
                    result[s.top()] += time - prevTime;
                }
                s.push(id);
                prevTime = time;
            } else {
                result[s.top()] += time - prevTime + 1;
                s.pop();
                prevTime = time + 1;
            }
        }
        return result;
    }
};

File: Java/Medium/Accepted/exclusive-time-of-functions_5.java
class Solution {
    public int[] exclusiveTime(int n, List<String> logs) {
        
        int[] result = new int[n];
        Stack<Integer> stack = new Stack<>();
        int prevTime = 0;
        for (String log : logs) {
            String[] s = log.split(":");
            int id = Integer.parseInt(s[0]);
            int time = Integer.parseInt(s[2]);
            if (s[1].equals("start")) {
                if (!stack.isEmpty()) {
                    result[stack.peek()] += time - prevTime;
                }
                stack.push(id);
                prevTime = time;
            } else {
                result[stack.pop()] += time - prevTime + 1;
                prevTime = time + 1;
            }
        }
        return result;
    }
}